src/game_api/rpc.cpp

```diff
@@ -573,34 +574,39 @@ std::vector<uint32_t> get_entities_by(std::vector<ENT_TYPE> entity_types, uint32
     std::vector<uint32_t> found;
     const std::vector<ENT_TYPE> proper_types = get_proper_types(std::move(entity_types));
 
-    auto push_matching_types = [&proper_types, &found](EntityList entities)
+    auto push_matching_types = [&proper_types, &found](const EntityList& entities)
     {
-        for (auto& item : entities)
+        for (auto& item : entities.entities())
         {
             if (entity_type_check(proper_types, item->type->id))
             {
                 found.push_back(item->uid);
             }
         }
     };
-    auto insert_all_uids = [&found](EntityList entities)
+    auto insert_all_uids = [&found](const EntityList& entities)
     {
-        found.insert(found.end(), entities.uid_begin(), entities.uid_end());
+        found.reserve(found.size() + entities.size);

```

Comment:
isn't this redundant? from what i've read, `insert` already does that if you give it a range

Reply:
I looked at the implementation in the MS STL and it doesn't do that. There's also nothing like that mandated by the standard.

Reply:
You sure? http://www.cplusplus.com/reference/vector/vector/insert/#complexity

Reply:
Not super sure, but I do not trust cplusplus.com, it has tons of incorrect information all over the place, sometimes literally contradicting the standard. cppreference.com does not name any complexity requirements in terms of allocations. And as I said, the MS STL does not do this either.

Reply:
I'll remove the two lines just so we get these changes in.

Reply:
well, it doesn't really change much if you leave or delete them anyway